### DISCUSSION TITLE
how to make transactions use web3-react

### DISCUSSION THREAD

**Discussion by WLyKan** [NoahZinsmeister/web3-react#744] [Q&A]
I read metamask doc, it says:
```js
const txHash = await ethereum.request({
  method: 'eth_sendTransaction',
  params: [transactionParameters],
});
```
how to perform this in web3-react?

**Comment by niZmosis** [comment]
Use the [ethers](https://docs.ethers.org/v5/api/signer/) library

**Reply by WLyKan** [reply]
after connect the MetaMask, I find the doc:
https://docs.metamask.io/guide/sending-transactions.html#example

I guess `ethereum.request` equal to `provider.request`, but got `TypeError: provider.request is not a function`

**Reply by Niwilai** [reply]
@WLyKan if you deconstruct the provider object from `const context = useWeb3React()` like so:
`const { isActive, account, chainId, provider } = context;`

Apparently the provider you are looking for is sitting on `provider`, so `provider.provider.request` will work.
